heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -161,6 +161,13 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
       Set<PackingPlan.ContainerPlan> containersAdded =
           scalableScheduler.get().addContainers(containersToAdd);
+      if (containersAdded.size() != containersToAdd.size()) {
```

Comment:
I am wondering if it is possible to move this check to RuntimeManagerRunner.java.

Reply:
how to?

Reply:
The function addContainers() above the if statement is in AuroraCLIController, the code looks like:

public Set<Integer> addContainers(Integer count) {
  ....
  return extractContainerIds(stderr.toString());
}

You should be able to check container added vs count in there before returning I think. Then no changes will be needed in this file.

Reply:
i think this check is for all schedulers, not only for aurora , as suggested by maosong

Reply:
kk. that could be useful.